admin/controllers/service-manager.js

```diff
@@ -13,374 +13,343 @@ const sortingAttributesService = require('../services/sorting-attributes.service
+    console.log('session:', req.session)
     try {
-      checkWindowData = await checkWindowDataService.sqlFindOneById(req.params.id)
-
-      const adminStartDate = moment(checkWindowData.adminStartDate, 'D MM YYYY').format('YYYY-MM-D')
-      const checkStartDate = moment(checkWindowData.checkStartDate, 'D MM YYYY').format('YYYY-MM-D')
-
-      checkWindowData = {
-        checkWindowId: req.params.id,
-        checkWindowName: checkWindowData.name,
-        adminStartDay: moment(checkWindowData.adminStartDate).format('D'),
-        adminStartMonth: moment(checkWindowData.adminStartDate).format('MM'),
-        adminStartYear: moment(checkWindowData.adminStartDate).format('YYYY'),
-        checkStartDay: moment(checkWindowData.checkStartDate).format('D'),
-        checkStartMonth: moment(checkWindowData.checkStartDate).format('MM'),
-        checkStartYear: moment(checkWindowData.checkStartDate).format('YYYY'),
-        checkEndDay: moment(checkWindowData.checkEndDate).format('D'),
-        checkEndMonth: moment(checkWindowData.checkEndDate).format('MM'),
-        checkEndYear: moment(checkWindowData.checkEndDate).format('YYYY'),
-        existingAdminStartDate: adminStartDate,
-        existingCheckStartDate: checkStartDate
+      const settingsRecord = await settingService.get() // settingDataService.sqlFindOne()
+      if (settingsRecord) {
+        settings = settingsRecord
+      } else {
+        settings = {
+          'questionTimeLimit': config.QUESTION_TIME_LIMIT,
+          'loadingTimeLimit': config.TIME_BETWEEN_QUESTIONS
+        }
       }
+    } catch (error) {
+      return next(error)
+    }
 
-      const currentDate = moment.utc(moment.now()).format('YYYY-MM-D')
-      if (moment(currentDate).isAfter(adminStartDate)) {
-        adminIsDisabled = 1
-      }
-      if (moment(currentDate).isAfter(checkStartDate)) {
-        checkStartIsDisabled = 1
-      }
+    try {
+      req.breadcrumbs(res.locals.pageTitle)
+      res.render('service-manager/check-settings', {
+        settings,
+        successfulPost,
+        breadcrumbs: req.breadcrumbs()
+      })
     } catch (error) {
-      return next()
+      next(error)
     }
-  }
+  },
+
+  /**
+   * Update time settings in DB.
+   * @param req
+   * @param res
+   * @param next
+   * @returns {Promise.<*>}
+   */
+  setUpdateTiming: async (req, res, next) => {
+    res.locals.pageTitle = 'Settings on pupil check'
+    let settings
 
-  res.render('service-manager/check-windows-form', {
-    breadcrumbs: req.breadcrumbs(),
-    error,
-    errorMessage,
-    checkWindowData,
-    successfulPost,
-    actionName,
-    urlActionName,
-    currentYear,
-    adminIsDisabled,
-    checkStartIsDisabled
-  })
-}
+    const settingsRecord = await settingService.get()
+    if (settingsRecord) {
+      settings = settingsRecord
+    } else {
+      settings = {}
+    }
+    settings.questionTimeLimit = Math.round(req.body.questionTimeLimit * 100) / 100
```

Comment:
should this be done in the settingService?

Reply:
@GuyHarwood correct and it will be in a separate PR since this is existing code.